packages/backend-common/src/database/connection.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import knex from 'knex';
+import { Config } from '@backstage/config';
+import { buildPgConnectionConfig } from './postgres';
+import { buildSqlite3ConnectionConfig } from './sqlite3';
+
+/**
+ * Creates a knex database connection
+ *
+ * @param config The database config
+ * @param dbName The optional database to connect to
+ */
+export function createDatabase(config: Config, dbName?: string) {
+  const knexConfig = getDatabaseConfig(config, dbName);
+  const database = knex(knexConfig);
+
+  database.client.pool.on('createSuccess', (_eventId: any, resource: any) => {
+    resource.run('PRAGMA foreign_keys = ON', () => {});
+  });
+
+  return database;
+}
+
+/**
+ * Gets a knex database config
+ *
+ * @param config The database config
+ * @param dbName The optional database to connect to
+ */
+export function getDatabaseConfig(
+  config: Config,
+  dbName?: string,
+): knex.Config {
+  const client = config.getString('client');
+
+  return {
+    client,
+    connection: buildDatabaseConnectionConfig(client, config, dbName),
+    useNullAsDefault: true,
```

Comment:
is this something that will always necessarily be true? not familiar with what this entails in practice for different databases

Reply:
I was debating having the client specific code create the entire config (client, connection, etc) vs combining it together in a common place. It might be better to have it all defined in the client specific builder.